thrifty-kotlin-codegen/src/test/kotlin/com/microsoft/thrifty/kgen/KotlinCodeGeneratorTest.kt

```diff
@@ -1346,8 +1351,25 @@ class KotlinCodeGeneratorTest {
         file.shouldCompile()
     }
 
+    @Test
+    fun `Files should add generated comment`() {
+        val thrift = """
+            |namespace kt test.comment
+            |
+            |const i32 FooNum = 42
+        """.trimMargin()
+
+        val file = generate(thrift) { emitFileComment(true) }
+                .single()
+        file.shouldCompile()
+
+        val lines = file.toString().split("\n")
+        lines[0] shouldBe "// Automatically generated by the Thrifty compiler; do not edit!"
+        lines[1] shouldContain "\\/\\/ Generated on: [0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}\\.\\S+".toRegex()
+    }
+
     private fun generate(thrift: String, config: (KotlinCodeGenerator.() -> KotlinCodeGenerator)? = null): List<FileSpec> {
-        val configOrDefault = config ?: { this }
+        val configOrDefault = config ?: { emitFileComment(false) }
```

Comment:
I can also default this to true but it's a bit annoying to handle the changing timestamp.